modules/system/checks.nix

```diff
@@ -31,6 +31,18 @@ let
     fi
   '';
 
+  primaryUser = ''
+    primaryUser=${escapeShellArg config.system.primaryUser}
```

Comment:
```suggestion
    primaryUser="${config.system.primaryUser}"
```

```
In /nix/store/726gp7jnx7blz46azrds8l28a5zqdqgg-darwin-system-25.05.0653ce2/activate line 63:
primaryUser=admin
^---------^ SC2209 (warning): Use var=$(command) to assign output (or quote to assign string).
 
For more information:
  https://www.shellcheck.net/wiki/SC2209 -- Use var=$(command) to assign outp...
```

Reply:
`escapeShellArg` will add some single quotes which will mess with the value of `primaryUser` I believe so best to leave it out here if we're manually quoting

Alternatively we could use `escapeShellArgAlways`:

https://github.com/nix-darwin/nix-darwin/blob/43975d782b418ebf4969e9ccba82466728c2851b/modules/services/github-runner/service.nix#L105-L108

Reply:
Could we also just do `escapeShellArg` and disable this warning? Although I don't know if that would be tangibly different from just doing what you suggested here.

Reply:
Feels like the linter is wrong here because it doesn't know the value came from a proper escape function. my +1 on silencing the linter. For what that's worth :)

Reply:
Silenced the linter 👍